tools/mafft/mafft.xml

```diff
@@ -1,305 +1,518 @@
+                                <param argument="--lexp" type="float" value="-0.1" label="Gap extension penalty at local pairwise alignment." help="-0.1 default value" />
+                            </when>
+                            <when value="--genafpair">
+                                <expand macro="weighti_param" />
+                                <param argument="--lop" type="float" value="-2.0" label="Gap opening penalty at local pairwise alignment" help="-2.00 default value" />
+                                <param argument="--lep" type="float" value="0.1" label="Offset value at local pairwise alignment" help="0.1 default value" />
+                                <param argument="--lexp" type="float" value="-0.1" label="Gap extension penalty at local pairwise alignment." help="-0.1 default value" />
+                                <param argument="--LOP" type="float" value="-6.00" label="Gap opening penalty to skip the alignment" help="-6.00 default value" />
+                                <param argument="--LEXP" type="float" value="0.00" label="Gap extension penalty to skip the alignment" help="0 default value" />
+                            </when>
+                            <when value="--fastapair">
+                                <expand macro="weighti_param" />
+                            </when>
+                        </conditional>
+                    </when>
+                    <when value="parttree">
+                        <conditional name="parttree_selection">
+                            <param name="parttree_option" type="select" label="Which distance for the fast tree-building method?">
+                                <option value="--parttree" selected="true">Fast tree-building method with the 6mer distance (--parttree)</option>
+                                <option value="--fastaparttree">Distances based on FASTA (--fastaparttree)</option>
+                                <option value="--dpparttree">Distances based on DP. (Needleman-Wunsch) (--dpparttree)</option>
+                            </param>
+                            <when value="--parttree">
+                                <expand macro="parttree_parameters" />
+                            </when>
+                            <when value="--fastaparttree">
+                                <expand macro="parttree_parameters" />
+                            </when>
+                            <when value="--dpparttree">
+                                <expand macro="parttree_parameters" />
+                            </when>
+                        </conditional>
+                    </when>
+                </conditional>
+            </when>
+        </conditional>
+        <section name="progressive_alignment_calculation" title="Progressive alignment calculation" expanded="true">
```

Comment:
Exposing these settings unconditionally makes no sense since they are part of the "flavour" definitions.
If a user selects a pre-defined flavour above and you're overwriting some of that flavour's settings here, it's not that flavour anymore.

@elischberg I've opened another PR against your branch that fixes this and a few other things.
Once that's merged you will have to regenerate all test data except for the first test (for which the settings here happen to be that of the flavour).
I guess it would be a good idea to generate all test data with mafft flavour commands on the command line to make sure the wrapper produces matching outputs.